src/cluster.c

```diff
@@ -126,6 +131,66 @@ static ConnectionType *connTypeOfCluster() {
 
     return connectionTypeTcp();
 }
+/* Cluster shards hash table, mapping shard id to list of nodes */
+dictType clusterSdsToListType = {
+        dictSdsHash,                /* hash function */
+        NULL,                       /* key dup */
+        NULL,                       /* val dup */
+        dictSdsKeyCompare,          /* key compare */
+        dictSdsDestructor,          /* key destructor */
+        dictListDestructor,         /* val destructor */
+        NULL                        /* allow to expand */
+};
+
+/* Aux field updater function prototype
+ * return 1 when the update is successful; 0 otherwise */
+typedef int (aux_updater) (clusterNode* n, void *value, int length);
+
+typedef struct {
+    char *field;
+    aux_updater *updater;
+    int present;
+} auxFieldUpdater;
+
+/* Assign index to each aux field */
+typedef enum {
+    shard_id_idx,
+} auxFieldIndex;
+
+/* Note that
```

Comment:
Would probably be good to add a comment block here, or earlier, defining what the aux fields are.

Reply:
Will do.